packages/belos/src/BelosBlockGmresSolMgr.hpp

```diff
@@ -813,6 +813,18 @@ bool BlockGmresSolMgr<ScalarType,MV,OP>::checkStatusTest() {
   // Basic test checks maximum iterations and native residual.
   maxIterTest_ = Teuchos::rcp( new StatusTestMaxIters<ScalarType,MV,OP>( maxIters_ ) );
 
+  // Perform sanity checking for flexible Gmres here.
+  // NOTE:  If the user requests that the solver manager use flexible GMRES, but there is no right preconditioner, don't use flexible GMRES.
+  //        Throw an error is the user provided a left preconditioner, as that is inconsistent with flexible GMRES.
+  if (isFlexible_ && Teuchos::is_null(problem_->getRightPrec())) {
+    isFlexible_ = false;
```

Comment:
What if users later introduce a right preconditioner?  The above code would permanently change `isFlexible_`, which users might find surprising.  We might instead want a "we are actually going to do Flexible GMRES" flag that is the disjunction of "the user asked for Flexible GMRES" (what `isFlexible_` currently means) and "we have a right preconditioner."